src/providers/pdal/qgspdalprovider.cpp

```diff
@@ -98,17 +100,21 @@ QgsPointCloudAttributeCollection QgsPdalProvider::attributes() const
   return QgsPointCloudAttributeCollection();
 }
 
-static QString _outEptDir( const QString &filename )
+static QString _outEptDir( const QString &uri )
 {
-  const QFileInfo fi( filename );
+  const QVariantMap parts = QgsProviderRegistry::instance()->decodeUri( "pdal", uri );
```

Comment:
Instead of the multiple QgsProviderRegistry::instance() calls introduced here, just make a static method which does this and use throughout this class.